back/engines/commercial/flag_inappropriate_content/app/services/flag_inappropriate_content/toxicity_detection_service.rb

```diff
@@ -15,29 +28,15 @@ def flag_toxicity!(flaggable, attributes: [])
         end
         return
       end
-      res = request_toxicity_detection texts
-      if toxicity_detected? res
-        flag_service.introduce_flag! flaggable, toxicity_label: (extract_toxicity_label(res) || 'without_label')
+      toxicity_attrs = texts.filter_map { |text| classify_toxicity(text) }
+      if toxicity_attrs.present?
+        flag_service.introduce_flag! flaggable, toxicity_attrs.first
```

Comment:
I was wondering if there could be another way to find the best response when multiple texts were detected as toxic (rare case). For example, using the length of the `ai_reason` part. Here, we just take the first response.